ocrd/ocrd/resource_manager.py

```diff
@@ -0,0 +1,221 @@
+            parent_dirs = [join(x, 'ocrd-resources') for x in [XDG_DATA_HOME, '/usr/local/share']]
+            for parent_dir in parent_dirs:
+                if Path(parent_dir).exists():
+                    all_executables += [x for x in listdir(parent_dir) if x.startswith('ocrd-')]
+        for this_executable in set(all_executables):
+            reslist = []
+            for res_filename in list_all_resources(this_executable):
+                res_name = Path(res_filename).name
+                resdict = [x for x in self.database.get(this_executable, []) if x['name'] == res_name]
+                if not resdict:
+                    self.log.info("%s resource '%s' (%s) not a known resource, creating stub in %s'" % (this_executable, res_name, res_filename, self.user_list))
+                    resdict = [self.add_to_user_database(this_executable, res_filename)]
+                resdict[0]['path'] = res_filename
+                reslist.append(resdict[0])
+            ret.append((this_executable, reslist))
+        return ret
+
+    def add_to_user_database(self, executable, res_filename, url=None):
+        """
+        Add a stub entry to the user resource.yml
+        """
+        res_name = Path(res_filename).name
+        res_size = Path(res_filename).stat().st_size
+        with open(self.user_list, 'r', encoding='utf-8') as f:
+            user_database = safe_load(f) or {}
+        if executable not in user_database:
+            user_database[executable] = []
+        if not self.find_resources(executable=executable, name=res_name, database=user_database):
+            resdict = {
+                'name': res_name,
+                'url': url if url else '???',
+                'description': 'Found at %s on %s' % (self.resource_dir_to_location(res_filename), datetime.now()),
+                'version_range': '???',
+                'size': res_size
+            }
+            user_database[executable].append(resdict)
+        with open(self.user_list, 'w', encoding='utf-8') as f:
+            f.write(RESOURCE_USER_LIST_COMMENT)
+            f.write('\n')
+            f.write(safe_dump(user_database))
+        return resdict
+
+    def find_resources(self, executable=None, name=None, url=None, database=None):
+        """
+        Find resources in the registry
+        """
+        if not database:
+            database = self.database
+        ret = []
+        if executable and executable not in database.keys():
+            return ret
+        for executable in [executable] if executable else database.keys():
+            for resdict in database[executable]:
+                if not name and not url:
+                    ret.append((executable, resdict))
+                elif url and url == resdict['url']:
+                    ret.append((executable, resdict))
+                elif name and name == resdict['name']:
+                    ret.append((executable, resdict))
+        return ret
+
```

Comment:
Also, it might still be helpful to encapsulate the default path here (think ocrd_tesserocr again):

```suggestion

    @property
    def default_resource_dir(self):
        return self.location_to_resource_dir('data')
```